File: sample/src/main/java/com/sunmi/samples/printerx/MainFragment.java
package com.sunmi.samples.printerx;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.sunmi.samples.printerx.databinding.FragmentMainBinding;

public class MainFragment extends Fragment {

    private FragmentMainBinding binding;

    public MainFragment() {
        super();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentMainBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.info.setOnClickListener(view1 -> ((MainActivity)getActivity()).switchFragment(InfoFragment.class, true));
        binding.ticket.setOnClickListener(view12 -> ((MainActivity)getActivity()).switchFragment(TicketFragment.class, true));
        binding.label.setOnClickListener(view13 -> ((MainActivity)getActivity()).switchFragment(LabelFragment.class, true));
        binding.file.setOnClickListener(view12 -> ((MainActivity)getActivity()).switchFragment(FileFragment.class, true));
        binding.cmd.setOnClickListener(view12 -> ((MainActivity)getActivity()).switchFragment(CmdFragment.class, true));
        binding.lcd.setOnClickListener(view14 -> ((MainActivity)getActivity()).switchFragment(LcdFragment.class, true));
        binding.cash.setOnClickListener(view14 -> ((MainActivity)getActivity()).switchFragment(CashFragment.class, true));

    }
}

File: sample-ktx/src/main/java/com/sunmi/samples/printerx/ui/main/MainFragment.kt
package com.sunmi.samples.printerx.ui.main

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.viewModels
import com.sunmi.samples.printerx.MainActivity
import com.sunmi.samples.printerx.R
import com.sunmi.samples.printerx.databinding.FragmentMainBinding
import com.sunmi.samples.printerx.ui.vm.PrinterViewModel

/**
 * 功能展示页
 */
class MainFragment : Fragment() {

    private lateinit var dataBinding : FragmentMainBinding

    companion object {
        fun newInstance() = MainFragment()
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        dataBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_main, container, false)
        return dataBinding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        dataBinding.info.setOnClickListener {
            (activity as MainActivity).switchFragment(InfoFragment.newInstance(),
                addToBackStack = true
            )
        }
        dataBinding.ticket.setOnClickListener {
            (activity as MainActivity).switchFragment(TicketFragment.newInstance(),
                addToBackStack = true
            )
        }
        dataBinding.label.setOnClickListener {
            (activity as MainActivity).switchFragment(LabelFragment.newInstance(),
                addToBackStack = true
            )
        }
        dataBinding.cmd.setOnClickListener {
            (activity as MainActivity).switchFragment(CommandFragment.newInstance(),
                addToBackStack = true
            )
        }
        dataBinding.file.setOnClickListener {
            (activity as MainActivity).switchFragment(FileFragment.newInstance(),
                addToBackStack = true
            )
        }
        dataBinding.cash.setOnClickListener {
            (activity as MainActivity).switchFragment(CashFragment.newInstance(),
                addToBackStack = true
            )
        }
        dataBinding.lcd.setOnClickListener {
            (activity as MainActivity).switchFragment(LcdFragment.newInstance(),
                addToBackStack = true
            )
        }
        val viewModel: PrinterViewModel by viewModels()
        context?.let { viewModel.initPrinter(it) }
    }

}